lib/threefry.ex

```diff
@@ -0,0 +1,90 @@
+defmodule PRNG.Threefry do
+  import Kernel, except: [unless: 2]
+  use Bitwise
+
+  defp is_threefry_key(key) when is_list(key) do
+    length(key) == 2 and Enum.all?(
+      fn x -> is_integer(x) end
+    )
+  end
+
+  def threefry_seed(seed) when is_integer(seed) do
+    k1 = seed >>> 32
+    k2 = seed &&& 0xFFFFFFFF
+    [k1, k2]
+  end
+
+
+  defp rotate_left(x, rot) when is_integer(x) do
+    nbits = 32
+    (x <<< rot) ||| (x >>> (nbits - rot))
+  end
+
+  defp apply_round([x1, x2], rot) do
+    y1 = x1 + x2
+    y2 = bxor(x1, rotate_left(x2, rot))
+
+    [y1, y2]
+  end
+
+  defp rolled_loop_step(i, [xs, _ks = [k1, k2, k3], _rotations = [r1, r2]]) do
+      [x1, x2] = Enum.reduce(r1, xs, fn rot, x ->
+        apply_round(x, rot)
+      end)
+
+      new_x = [x1+k1, x2+k2+i+1]
+      new_k = [k2, k3, k1]
+      new_r = [r1, r2]
+
+      [new_x, new_k, new_r]
+  end
+
+  def threefry2x32_20([key1, key2] = ks, [_x1, _x2] = xs) do
+    rotations = [[13, 15, 26, 6],
+               [17, 29, 16, 24]]
+
+    xs = Enum.zip_with([xs, ks], fn [x, k] -> x+k end)
+    ks = [key1, key2, bxor(key1, key2) |> bxor(0x1BD11BDA)]
+
+    state = [xs, ks, rotations]
+
+
+    [final_xs, _final_ks, _final_rotations] = Enum.reduce(
+      Enum.to_list(0..5), state, fn i, state ->
+        rolled_loop_step(i, state)
+    end)
+
+    final_xs |> Enum.map(fn x -> x &&& 0xFFFFFFFF end)

```

Comment:
Ranges are enumerable so no need to turn then into lists
```suggestion
    0..5
    |> Enum.reduce(state, &rolled_loop_step/2)
    |> hd()
    |> Enum.map(&band(&1, 0xFFFFFFFF))
```